xarray/tests/test_coding_times.py

```diff
@@ -1311,12 +1311,16 @@ def test_roundtrip_timedelta64_nanosecond_precision_warning() -> None:
         f"Timedeltas can't be serialized faithfully with requested units {units!r}. "
         f"Resolution of {needed_units!r} needed. "
     )
-    encoding = dict(_FillValue=20, units=units)
+    encoding = dict(dtype=np.int64, _FillValue=20, units=units)
```

Comment:
Can we make the same modification in the `test_roundtrip_datetime64_nanosecond_precision_warning` test to get coverage there too?